torchci/components/TorchAgentPage.tsx

```diff
@@ -106,6 +106,10 @@ export const TorchAgentPage = () => {
   const [error, setError] = useState("");
   const [debugVisible, setDebugVisible] = useState(false);
   const [currentSessionId, setCurrentSessionId] = useState<string | null>(null);
+  const [hasInsufficientPermissions, setHasInsufficientPermissions] =
+    useState(false);
+  const [isCheckingPermissions, setIsCheckingPermissions] = useState(false);
+  const [hasCheckedPermissions, setHasCheckedPermissions] = useState(false);
```

Comment:
do we need three separate boolean flags for this? seems overcomplicated, maybe just have a single enum for permissions state machine?

Reply:
made an enum